packages/languages/src/browser/window-impl.ts

```diff
@@ -56,8 +56,17 @@ export class WindowImpl implements Window {
         return {
             append: outputChannel.append.bind(outputChannel),
             appendLine: outputChannel.appendLine.bind(outputChannel),
-            show: function (preserveFocus?: boolean): void {
-                // no-op
+            show: async (preserveFocus?: boolean) => {
+                outputChannel.setVisibility(true);
+                const options = Object.assign({
+                    preserveFocus: false,
+                }, { preserveFocus });
+                const activate = !options.preserveFocus;
+                const reveal = options.preserveFocus;
+                this.outputContribution.openView({ activate, reveal });
```

Comment:
really reveal? if it's already opened but hidden behind another widget, I personally wouldn't like it to bother me. e.g. I'd prefer to see the watcher output in terminal.

Reply:
It depends how the language client is configured, by default it won't be displayed.